core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -485,6 +486,12 @@ class GroupCoordinator(val brokerId: Int,
       if (member.awaitingSyncCallback != null) {
         member.awaitingSyncCallback(member.assignment, errorCode)
         member.awaitingSyncCallback = null
+
+        // reset the session timeout for members after propagating the member's
```

Comment:
The comments here are a bit misleading, could we change it to:

"... assignment. This is because if any member's session expired while we were still awaiting either the leader sync group or the storage callback, its expiration will be ignored but future heartbeat expectations will not be scheduled."